src/main/java/note/NoteList.java

```diff
@@ -178,20 +197,58 @@ public void selectSearchMethod(String userInput) throws InvalidNoteException {
         }
     }
 
+    public void selectSearchMethod(String[] userInputInArray)
+            throws InvalidNoteException, NumberFormatException {
+        if (userInputInArray[1].equals("keyword")) {
+            if (userInputInArray.length == 2) {
+                keywordSearch();
+            } else {
+                keywordSearch(userInputInArray[2]);
+            }
+        } else if (userInputInArray[1].contains("index")) {
+            if (userInputInArray.length == 2) {
+                indexSearch();
+            } else {
+                indexSearch(userInputInArray[2]);
+            }
+        } else {
+            throw new InvalidNoteException(INVALID_NOTE_SEARCH_MESSAGE);
+        }
+    }
+
+    public void keywordSearch(String userInput) {
+        ui.printSelectedNote(this.searchNoteUsingTitle(userInput, this));
+    }
+
     public void keywordSearch() {
         ui.printNoteSearchKeyWordInstructions();
         String keywords = ui.readUserInput();
-        //System.out.println(keywords);
         ui.printSelectedNote(this.searchNoteUsingTitle(keywords, this));
     }
 
-    public void indexSearch() {
+    public void indexSearch(String userInput) throws NumberFormatException {
+        ui.printNoteSearchSceneIndexInstructions();
+        int sceneIndex = Integer.parseInt(userInput);
+        ui.printSelectedNote(this.searchNotesUsingSceneIndex(sceneIndex, this));
+    }
+
+    public void indexSearch() throws NumberFormatException {
         ui.printNoteSearchSceneIndexInstructions();
         int sceneIndex = Integer.parseInt(ui.readUserInput());
         ui.printSelectedNote(this.searchNotesUsingSceneIndex(sceneIndex, this));
     }
 
-    public void openNoteDirectly() throws IndexOutOfBoundsException  {
+    public void openNoteDirectly(String index) throws IndexOutOfBoundsException, NumberFormatException {
+        ui.printNoteOpenInstructions();
+        //here the index is not scene index, it is the index in the list
```

Comment:
I think our comments follow the format like // Here the index..., with space after // and start with capital letter.